src/mono/netcore/sample/iOS/Program.cs

```diff
@@ -48,6 +48,9 @@ public static async Task Main(string[] args)
         }
 
         Console.WriteLine("Done!");
-        await Task.Delay(-1);
+
+        await Task.Delay(5000);
+
+        return 42;
```

Comment:
Is there some reason we can't return 0 on success, as is the typical convention?

Reply:
I have the same question as Nathan.

Reply:
No reason.  The return code was made up and I agree it should be 0.

Reply:
The reason of adding non-zero exit code is to test the support of the xharness' `--expected-exit-code` option, which was aligned with the android part. 
Initially it was assumed that the iOS sample could run like other library test app but CI runs showed that it could lead to some crash due to the xharness expected the test-related things like test output result etc. So `--expected-exit-code` in this case added the capability to avoid such crashes and let xharness know that the sample app run as expected.
On the other hand, recently the xharness has got the separate command just to run iOS application on simulator, which does not require a test runner and relies on the app's exit code. So I assume that returning 0 should work.

Reply:
Now it returns 0.

Reply:
There is another reason as well - it is hard to detect exit code of iOS apps. We currently don't have a proper way other than scanning some system logs and this solution might not work across different iOS versions. We only implemented it this fast to support the scenario with a proper support planned in the future (if there is one possible).

So it makes sense to return non-0 and verify it here in case it breaks in the future.

Reply:
Afaik, Android does the same?

Reply:
> Afaik, Android does the same?  

Yeah, Android sample already returns 42

Reply:
@premun Let CI finish the round with zero exit code so we could see some related results.

Reply:
FYI: https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-44462-merge-30daa21b383d4a538c/HelloiOS/console.67fdad7f.log?sv=2019-07-07&se=2020-12-29T10%3A30%3A52Z&sr=c&sp=rl&sig=4UwbLxAy1LoC2Bj05Wl3F6FqBesGBET0xx3wyPGawwA%3D

```
[05:38:30] dbug: 05:38:30.2730250 Xamarin.Hosting: Launched net.dot.HelloiOS with pid 51525
[05:38:47] dbug: 05:38:47.3250590 Xamarin.Hosting: Simulated process has exited.
[05:38:47] dbug: 05:38:47.3308300 Process mlaunch exited with 0
[05:38:47] dbug: 05:38:47.3455210 App run ended with 0
[05:38:47] info: Application has finished with exit code: 0
XHarness exit code: 0
```

Reply:
Yes, this will work but I wouldn't advise it. It won't detect anything in the logs (since it's a naive full-text search) so it will assume 0 was returned.

If new iOS version changes this and the detection will stop working, we won't notice even if the app returns non-0 and then the app can break and we still won't know. So I think it's better to return non-0 and validate against that.

If we figure out a better and more reliable way, we can put 0 here but we don't have that now.

Reply:
Yeah, your words make sense, no issue with bringing non-zero exit code back. At least we verified this case :)

Reply:
Is it possible that XHarness's exit code could be 42? If not, 42 is okie.

Reply:
I think another way we could make it more clear is to have the 42 return code only for CI testing, e.g. something like this:

```csharp
#if CI_TEST
   await Task.Delay(5000);
   return 42;
#else
  await Task.Delay(-1);
#endif 
```

and then setting the CI_TEST define based on whether we're building for CI.

Reply:
Added the `CI_TEST` define, thanks!